jaxrs/src/main/java/org/killbill/billing/jaxrs/resources/AccountResource.java

```diff
@@ -688,7 +688,7 @@ public Response payAllInvoices(@PathParam("accountId") final String accountId,
                                            invoice.getBalance() : remainingRequestPayment;
             if (amountToPay.compareTo(BigDecimal.ZERO) > 0) {
                 final UUID paymentMethodId = externalPayment ? null : account.getPaymentMethodId();
-                createPurchaseForInvoice(account, invoice.getId(), amountToPay, paymentMethodId, externalPayment, pluginProperties, callContext);
+                createPurchaseForInvoice(account, invoice.getId(), amountToPay, paymentMethodId, externalPayment, UUIDs.randomUUID().toString(), UUIDs.randomUUID().toString(), pluginProperties, callContext);
```

Comment:
@pierre 
Ok sure! The DefaultControlInitiated will remain as-is.
I've updated the code with the removal of the not null verification of the transactionExternalKey (All integration tests are running fine) and I've also merged with the latest changes in work-for-release-0.17.x

Thanks!